### PR TITLE
feat(text): tagName prop to override to tag

### PR DESCRIPTION
**Pull request by Coread**
# Description

Add the prop tagName to allow the overriding of the tag used for the text component. The tagName prop is optional and if not supplied, the default tagName for each type will be used.

# Links

*Links to relevent resources.*

### CODE REVIEW

File: src/components/Text/Text.types.ts [hunk 1]
```diff
@@ -20,6 +20,10 @@ export interface Props {
    * Custom style for overriding this component's CSS.
    */
   style?: CSSProperties;
+  /**
+   * Override the tag used to surround the text
+   */
+  tagName?: keyof JSX.IntrinsicElements;
```
Comment (cscheffauer) [inline src/components/Text/Text.types.ts:26]: for better typing and avoiding mistakes by the consumer, i think u could type it more precise like:
`tagName?: keyof JSX.IntrinsicElements['h1' | 'h2' | 'h3' | ...];`

File: src/components/Text/Text.unit.test.tsx [hunk 2]
```diff
@@ -60,6 +60,21 @@ describe('Text', () => {
     expect(container).toMatchSnapshot();
   });
 
+  it('should match snapshot with type and tagName override', async () => {
+    expect.assertions(1);
+
+    const texts = Object.values(TYPES).map((type, index) => {
+      return (
+        <Text key={index} type={type as FontStyle} tagName="div">
```
Comment (cscheffauer) [inline src/components/Text/Text.unit.test.tsx:68]: div should not be allowed as a tagName - i think we should avoid unit tests which are actually breaking the component. instead would be nice to use different "allowed tagNames"

File: src/components/Text/Text.stories.tsx [hunk 3]
```diff
@@ -41,10 +41,27 @@ Types.parameters = {
...
+TypesWithTagName.argTypes = { ...argTypes };
+delete TypesWithTagName.argTypes.type;
+
+TypesWithTagName.parameters = {
+  variants: [
+    ...Object.values(TYPES).map((type) => {
+      return {
+        type,
+        children: type,
+        tagName: 'div',
```
Comment (cscheffauer) [inline src/components/Text/Text.stories.tsx:55]: same as below - div should not be allowed as a tagName for text, therefore having it here for the story is not recommended.

**Review by cscheffauer — CHANGES_REQUESTED** [review]
some comments on improving it, otherwise looking good

File: src/components/Text/Text.types.ts (hunk 1 above)
Comment (Coread) [reply src/components/Text/Text.types.ts]: changed

File: src/components/Text/Text.unit.test.tsx (hunk 2 above)
Comment (Coread) [reply src/components/Text/Text.unit.test.tsx]: changed, now testing all allowed tag names

File: src/components/Text/Text.stories.tsx (hunk 3 above)
Comment (Coread) [reply src/components/Text/Text.stories.tsx]: updated to use small here

**Review by cscheffauer — APPROVED** [review]

**Pull request merged**